From 714125bc7da655244e91c97da485da7a3838ab05 Mon Sep 17 00:00:00 2001
From: Frank Binns <frank.binns@imgtec.com>
Date: Thu, 19 Mar 2015 14:07:23 +0000
Subject: [PATCH] dri: Add MT12 and MT21 DRI fourcc

Add MT12 and MT21 DRI fourcc that corresponds to
DRM_FORMAT_MT12 and DRM_FORMAT_MT21 respectively.
---
 include/GL/internal/dri_interface.h | 2 ++
 src/egl/drivers/dri2/egl_dri2.c     | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/GL/internal/dri_interface.h b/include/GL/internal/dri_interface.h
index 35152c7..ab98427 100644
--- a/include/GL/internal/dri_interface.h
+++ b/include/GL/internal/dri_interface.h
@@ -1080,6 +1080,8 @@ struct __DRIdri2ExtensionRec {
 #define __DRI_IMAGE_FOURCC_NV12		0x3231564e
 #define __DRI_IMAGE_FOURCC_NV16		0x3631564e
 #define __DRI_IMAGE_FOURCC_YUYV		0x56595559
+#define __DRI_IMAGE_FOURCC_MT12		0x3231544D
+#define __DRI_IMAGE_FOURCC_MT21		0x3132544D
 
 
 /**
diff --git a/src/egl/drivers/dri2/egl_dri2.c b/src/egl/drivers/dri2/egl_dri2.c
index 43ab790..d7b3990 100644
--- a/src/egl/drivers/dri2/egl_dri2.c
+++ b/src/egl/drivers/dri2/egl_dri2.c
@@ -1785,6 +1785,8 @@ dri2_check_dma_buf_format(const _EGLImageAttribs *attrs)
    case DRM_FORMAT_NV21:
    case DRM_FORMAT_NV16:
    case DRM_FORMAT_NV61:
+   case DRM_FORMAT_MT12:
+   case DRM_FORMAT_MT21:
       plane_n = 2;
       break;
    case DRM_FORMAT_YUV410:
-- 
1.9.1

